pkg/features/features.go

```diff
@@ -19,6 +19,7 @@ var (
 	Flags = make(map[string]FeatureFlag)
 )
 
+// registerFeature global registers and returns a new feature flag that can be changed from the default state regardless of build.
 func registerFeature(name, envVar string, defaultValue bool) FeatureFlag {
```

Comment:
Should it be the other way around: `registerFeature` registers a legacy feature that's not changeable, whereas a new function is introduced to register a feature mutable in runtime? This way, there's no need to modify existing callers of `registerFeature`, and use of the new function can be gradually introduced.

Reply:
I think it is better to make the default `registerFeature` be the function that makes a runtime-changeable feature, and the special case function be the one that makes a build-time fixed flag. This will get people used to thinking that features are changeable by default.

The "blast area" of `registerFeature` is very small, and this PR already makes the necessary changes.

The old, immutable flags will gradually fall away, and new immutable ones should be special cases, or for early stages of a feature's development, where we don't want to risk incomplete features leaking out into the wild.

I can foresee the progression as:
1. Create flag as immutable when feature is in very early stages, so as not to delay releases.
2. Switch flag, with same name, to mutable, when it is safe to allow testing in customer environments.
3. Some flags will eventually be removed, because the feature is stable enough to be "always on"; other flags may continue indefinitely for special use cases.

Reply:
That's reasonable, therefore I won't push on my suggestion here.